backend_new/src/services/application.service.ts

```diff
@@ -195,68 +211,343 @@ export class ApplicationService {
   /*
     this will return 1 application or error
   */
-  async findOne(applicationId: string) {
-    const rawApplication = await this.prisma.applications.findFirst({
+  async findOne(applicationId: string): Promise<Application> {
+    const rawApplication = await this.findOrThrow(
+      applicationId,
+      ApplicationViews.details,
+    );
+
+    return mapTo(Application, rawApplication);
+  }
+
+  /*
+    this will create an application
+  */
+  async create(
+    dto: ApplicationCreate,
+    forPublic: boolean,
+  ): Promise<Application> {
+    // TODO: perms https://github.com/bloom-housing/bloom/issues/3445
+
+    const listing = await this.prisma.listings.findUnique({
       where: {
-        id: {
-          equals: applicationId,
-        },
+        id: dto.listings.id,
       },
-      include: {
-        userAccounts: true,
-        applicant: {
-          include: {
-            applicantAddress: true,
-            applicantWorkAddress: true,
-          },
-        },
-        applicationsMailingAddress: true,
-        applicationsAlternateAddress: true,
-        alternateContact: {
-          include: {
-            address: true,
-          },
-        },
-        accessibility: true,
-        demographics: true,
-        householdMember: {
-          include: {
-            householdMemberAddress: true,
-            householdMemberWorkAddress: true,
-          },
-        },
-        preferredUnitTypes: true,
+    });
+    if (forPublic) {
+      // if its a public submission
+      if (
+        listing &&
+        listing.applicationDueDate &&
```

Comment:
minor, but you could accomplish this with a typescript null check `listing?.applicationDueDate` instead of `listing && listing.applicationDueDate`

Reply:
nice call out, updated!